app/models/health/careplan_saver.rb

```diff
@@ -31,11 +31,25 @@ def update
       begin
         @careplan.class.transaction do
           if should_really_create_qa
-            @qualifying_activity.activity = :pctp_signed
-            @qualifying_activity.mode_of_contact = :other
-            @qualifying_activity.reached_client = :collateral
-            @qualifying_activity.mode_of_contact_other = 'On-line'
-            @qualifying_activity.reached_client_collateral_contact = 'On-line Signature'
+            signature_date = @careplan.provider_signed_on
+            @qualifying_activity.date_of_activity = signature_date
+
+            case @careplan.provider_signature_mode
```

Comment:
Might be worth throwing a `to_s` on here just in case this doesn't always round-trip to the DB in the future.